rllib/algorithms/tests/test_callbacks.py

```diff
@@ -74,6 +99,40 @@ def setUpClass(cls):
     def tearDownClass(cls):
         ray.shutdown()
 
+    def test_on_workers_recreated_callback(self):
+        config = (
+            APPOConfig()
+            .environment(CartPoleCrashing)
+            .callbacks(OnWorkerCreatedCallbacks)
+            .rollouts(num_rollout_workers=2)
+            .fault_tolerance(recreate_failed_workers=True)
+        )
+
+        for _ in framework_iterator(config, frameworks=("tf2", "torch")):
+            algo = config.build()
+            original_worker_ids = algo.workers.healthy_worker_ids()
+            for id_ in original_worker_ids:
+                self.assertTrue(algo._counters[f"worker_{id_}_recreated"] == 0)
+
+            # After building the algorithm, we should have 2 healthy (remote) workers.
+            self.assertTrue(len(original_worker_ids) == 2)
+
+            # Train a bit (and have the envs/workers crash a couple of times).
+            for _ in range(3):
+                algo.train()
+
+            # After training, each new worker should have been recreated.
+            worker_ids_2 = algo.workers.healthy_worker_ids()
+            for id_ in worker_ids_2:
+                # A newly created worker: It's recreated counter should be 1.
+                if id_ not in original_worker_ids:
+                    self.assertTrue(algo._counters[f"worker_{id_}_recreated"] == 1)
```

Comment:
Would this not become flakey in case a worker gets created >1 times? We iterate three times so i'd expect that after each iteration, workers might be recreated.

Reply:
I think it's fine a recreated worker gets a new ID, hence us checking, which are the new healthy worker IDs right before this loop.